utils/hoursUtils.js

```diff
@@ -0,0 +1,138 @@
+                isOpenNow: false,
+                openingSoon: undefined,
+                closingSoon: undefined,
+                hours: hoursSummary
+            }
+        }
+    },
+    
+    transformHours: function(time) {
+        if(!isNaN(time)) {
+            const minutes = time[2] + time[3]
+            var hours = time[0] + time[1]
+            const ampm = hours >= 12 ? "pm" : "am"
+            hours = (hours % 12) || 12
+            return `${hours}:${minutes} ${ampm}`
+        } else {
+            return time
+        }
+    },
+
+    checkIsOpenNow: checkIsOpenNow
+    
+}
+
+getHoursSummary = function(hoursFields) {
+    const formattedHours = hoursFields.map(field => 
+        {
+            const closeTimeDigits = field.close_time_digits
+            const hoursObject = {
+                day: field.open_weekday,
+                day_digit: field.open_weekday_digit,
+                openTime: field.open_time,
+                open_time_digits: field.open_time_digits,
+                closeTime: field.close_time,
+                close_time_digits: closeTimeDigits ? closeTimeDigits : undefined,
+                hoursSummary: closeTimeDigits ? `${field.open_time} - ${field.close_time}` : "Open 24 hours"
+            }
+            return hoursObject
+        }
+    )
+
+    const weeklySchedule = [
+        getDaySchedule("sunday", 0, formattedHours),
+        getDaySchedule("monday", 1, formattedHours),
+        getDaySchedule("tuesday", 2, formattedHours),
+        getDaySchedule("wednesday", 3, formattedHours),
+        getDaySchedule("thursday", 4, formattedHours),
+        getDaySchedule("friday", 5, formattedHours),
+        getDaySchedule("saturday", 6, formattedHours)
+    ]
+    return weeklySchedule
+
+}
+
+getDaySchedule = function(weekday, dayDigit, formattedHours) {
+    const hours = (formattedHours.find( ({ day_digit }) => day_digit === dayDigit ) || {})
+
+    return {
+        day: weekday,
+        dayDigit: dayDigit,
+        openTime: hours.openTime,
+        closeTime: hours.closeTime,
+        isToday: checkIsToday(dayDigit),
+        hoursSummary: hours.hoursSummary || "Closed"
+    }
+}
+
+checkIsToday = function(dayDigit) {
+    const todayDigit = getDatetimeNow().getDay()
+    return todayDigit === dayDigit
+}
+
+parseTodayHours = function(todayHours, datetime) {
+    const opening = todayHours.open_time_digits
+    const closing = todayHours.close_time_digits
+    
+    const nowHours = addZero(datetime.getHours())
+    const nowMinutes = addZero(datetime.getMinutes())
+    const nowTime = nowHours + "" + nowMinutes
+
+    const isOpenNow = checkIsOpenNow(opening, closing, nowTime)
+    const openingSoon = checkIsOpeningSoon(opening, nowTime)
+    const closingSoon = checkIsClosingSoon(closing, nowTime)
+    return {
+        isOpenNow: isOpenNow,
+        openingSoon: openingSoon,
+        closingSoon: closingSoon,
+    }
+}
+
+function checkIsOpenNow(opening, closing, nowTime) {
```

Comment:
After thinking about it. this is probably the form for function declaration you'll want to use in general.

You won't need to worry about declaring functions in a particular order if you do so.

Reply:
ah, okay - can update that

Reply:
updated for the rest in this file